cowprotocol/cowswap
# fix: uploading appData too late

# Summary

We were trying to decode the TWAP transactions that our users were doing. It turned out our decoding was working for some, but not for other txs. To present a good decoding of the TWAP transaction we need the appData as it contains information for the slippage, fee etc. It turned out that some of the txs were not having app data. This is due to the fact that we are trying to decode the transaction before signing, but at this point no appData was being uploaded to the server.

Moving the uploadAppData call before the call for signing the tx solves this problem and makes sure that appData is available even before signing.

# To Test
Open the network tab, create a twap order, hit the "place twap order" button. You should see that the app_data is being uploaded even before there is a signature from the user.
